### PR TITLE
Update environment to add multiyaml file support

### PR DESCRIPTION
**Pull request by mharshe**
- Adds support to load information from additional yaml files with specific root node name into the environment. Works with DDS, ZMQ & YAML environment.
- Updates API to support this additional node name as parameter.
- In case of ROS, root node name is used to add namespaces. Following ROS convention, empty node name implies relative key (key = "camera", rootnode = "" --> `nodeHandle.getParam("camera" ...)` ). Otherwise, `key = "camera", rootNode="robot"` ---> `nodeHandle.getParam("/robot/camera", ...)`

### CODE REVIEW

File: core/modules/ros_mdlw/kpsr_ros_core/src/ros_env.cpp [hunk 1]
```diff
@@ -19,40 +19,84 @@
...
+#include <stdexcept>
 
 kpsr::ros_mdlw::RosEnv::RosEnv(ros::NodeHandle * nodeHandle) {
 	this->nodeHandle = nodeHandle;
 }
 
-void kpsr::ros_mdlw::RosEnv::getPropertyString(const std::string key, std::string & value) {
-    nodeHandle->getParam(key, value);
+void kpsr::ros_mdlw::RosEnv::getPropertyString(const std::string& key, std::string & value, const std::string& rootNode) {
+    if (rootNode.size() == 0) {
```
Comment (pablogh-2000) [inline core/modules/ros_mdlw/kpsr_ros_core/src/ros_env.cpp]: maybe we can create a small function or inline macro for this? I.e., calculate key for key + rootNode?

File: core/tests/dds_core_tests/src/dds_env_test.cpp [hunk 2]
```diff
@@ -42,11 +42,11 @@ TEST(DdsEnvironmentTest, DdsEnvironmentTest) {
     dds::sub::DataReader<kpsr_dds_core::DDSEnvironmentData> datareader2(sub, topic);
 
     kpsr::YamlEnvironment yamlEnvSub;
-    yamlEnvSub.reload("greeting: \"hello\"\ntest: 123\niteration: -1\nkpsr_dds_env_key: \"test\"");
+    yamlEnvSub.updateConfiguration("greeting: \"hello\"\ntest: 123\niteration: -1\nkpsr_dds_env_key: \"test\"");
```
Comment (pablogh-2000) [inline core/tests/dds_core_tests/src/dds_env_test.cpp]: We need much more testing please. In all middlewares and testing the new root node, loading multiple files. etc.

File: core/modules/ros_mdlw/kpsr_ros_core/src/ros_env.cpp (hunk 1 above)
Comment (mharshe) [reply core/modules/ros_mdlw/kpsr_ros_core/src/ros_env.cpp]: Done

File: core/tests/dds_core_tests/src/dds_env_test.cpp (hunk 2 above)
Comment (mharshe) [reply core/tests/dds_core_tests/src/dds_env_test.cpp]: Added additional tests for dds, zmq and yaml environments.

**Pull request merged**